ee/tables/nix_env/upgradeable/upgradeable.go

```diff
@@ -0,0 +1,144 @@
+//go:build linux
+// +build linux
+
+package nix_env_upgradeable
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/kolide/launcher/ee/dataflatten"
+	"github.com/kolide/launcher/ee/tables/dataflattentable"
+	"github.com/kolide/launcher/ee/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+const allowedCharacters = "0123456789-"
+
+type Table struct {
+	logger    log.Logger
+	tableName string
+	execCC    allowedcmd.AllowedCommand
+}
+
+func TablePlugin(logger log.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("uid"),
+	)
+
+	t := &Table{
+		logger:    logger,
+		tableName: "kolide_nix_upgradeable",
+		execCC:    allowedcmd.NixEnv,
+	}
+
+	return table.NewPlugin(t.tableName, columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	for _, uid := range tablehelpers.GetConstraints(queryContext, "uid", tablehelpers.WithAllowedCharacters(allowedCharacters)) {
```

Comment:
This won't trigger a warning if UID is missing. Do we need that?

Reply:
I added an error for this.